jarviscli/plugins/stt.py

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env python3
+import speech_recognition as sr
+import pyttsx3
+import subprocess
+from plugin import plugin
+@plugin()
+def hear(jarvis, s):
+    r = sr.Recognizer()
+    engine = pyttsx3.init()
+    listen = False
+
+    while listen is False:
+        try:
+            with sr.Microphone() as source:
+                subprocess.call('reset', shell=False)
+                print("Say listen to start listening")
+                r.adjust_for_ambient_noise(source)
+                audio = r.listen(source)
+                pinger = r.recognize_google(audio)
+            try:
+                if (pinger.lower() == "listen"):
+                    listen = True
+                    print("Voice mode activated. Say something!")
+                    break
+                else:
+                    continue
+            except LookupError:
+                continue
+        except sr.UnknownValueError:
+            continue
+
+    while listen is True:
+        print("Say somthing")
+        try:
+            with sr.Microphone() as source:
+                r.adjust_for_ambient_noise(source)
+                audio = r.listen(source)
+                pinger = r.recognize_google(audio).lower()
+
+            if (pinger == "stop"):
+                listen = False
+                print("Listening stopped.")
+                break
+            else:
+                print(pinger)
+                if pinger[0:5] == "go to":
+                    subprocess.call(["wmctrl", "-a", pinger[7:]], shell=False)
+                elif pinger[0:9] == "workspace":
+                    pinger = pinger[10:len(pinger)]
+                    if pinger == 'one':
+                        pinger = 1
+                    num = str(int(pinger) - 1)
+                    subprocess.call(["wmctrl", "-s", num], shell=False)
+                elif pinger[0:4] == "open":
+                    pinger = pinger[5:]
+                    subprocess.call(["nohup", pinger], shell=False)
+                else:
+                    line = pinger
+                    jarvis = jarvis._jarvis
```

Comment:
Just testet again - Really work amazing!

Told you to use jarvis._jarvis here. This created a variable-conflict - if this line is executed a second time jarvis allready is an instance of Jarvis - which does not have the attribute "_jarvis". So should be renamed something like

```_jarvis = jarvis._jarvis
line = _jarvis.precmd(line)
stop = _jarvis.onecmd(line)
stop = _jarvis.postcmd(stop, line)
```

Reply:
Ok. Fine. :) I will try to fix pyaudio problem.